oggm/tests/test_models.py

```diff
@@ -350,6 +351,49 @@ def test_past_mb_model(self, hef_gdir):
                                                     years=mbdf.index.values)
         assert_allclose(s, mbdf['MY_MB'])
 
+    def test_prcp_fac_update(self, hef_gdir):
+
+        gdir = hef_gdir
+        init_present_time_glacier(gdir)
+
+        mb_mod = massbalance.PastMassBalance(gdir, bias=0,
+                                             check_calib_params=False)
+        # save old precipitation time series
+        prcp_old = mb_mod.prcp
+        prcp_fac_old = cfg.PARAMS['prcp_scaling_factor']
+        # basic checks
+        assert mb_mod.prcp_fac == prcp_fac_old
+        assert mb_mod._prcp_fac == prcp_fac_old
+
+        # increase prcp by factor of 10
+        factor = 10
+        mb_mod.prcp_fac = factor
+        assert mb_mod.prcp_fac == factor
+        assert mb_mod._prcp_fac == factor
+        prcp_new = mb_mod.prcp
+        assert_allclose(prcp_new, prcp_old * factor / prcp_fac_old)
+
+        # check if it gets back to the old prcp time series
+        mb_mod.prcp_fac = prcp_fac_old
+        assert mb_mod.prcp_fac == prcp_fac_old
+        assert mb_mod._prcp_fac == prcp_fac_old
+        assert_allclose(mb_mod.prcp, prcp_old)
+
+        with pytest.raises(InvalidParamsError):
+            mb_mod.prcp_fac = -100
+
+        # check if an error is raised if check_calib_params
+        # is true
+        mb_mod = massbalance.PastMassBalance(gdir, bias=0,
+                                             check_calib_params=True)
+        with pytest.raises(InvalidWorkflowError):
+            mb_mod.prcp_fac = 10
+
+        # normally this should also be implemented for mu_star ...
+        # with pytest.raises(InvalidWorkflowError):
```

Comment:
Yes, I arguably thought that the chances that someone changes mu* happens by mistake are low, but it's also OK to be careful. Leave this bit here so that we remember